pkg/cmd/roachtest/cluster.go

```diff
@@ -2846,6 +2852,38 @@ func (c *clusterImpl) StopGrafana(ctx context.Context, l *logger.Logger, dumpDir
 	return roachprod.StopGrafana(ctx, l, c.name, dumpDir)
 }
 
+// AddGrafanaAnnotation creates a grafana annotation for the centralized grafana instance.
+func (c *clusterImpl) AddGrafanaAnnotation(
+	ctx context.Context, l *logger.Logger, req grafana.AddAnnotationRequest,
+) error {
+	// If grafanaTags is empty, then grafana is not set up for this cluster.
+	if len(c.grafanaTags) == 0 {
+		return errors.New("Grafana is not available for this cluster")
```

Comment:
Should this just be a log message? Otherwise it will require every caller to guard against the cloud type before calling this function.

Reply:
I actually went back and forth on this. My rationale was that I couldn't think of a case where the caller *would* ever want to fail a test because a grafana annotation failed, but if they did they'd probably want to fail because Grafana wasn't available too. So I left it up to the caller to decide if they want to log or fail the error.

But given that thinking of "I can't think of why we'd ever want to fail a test because an annotation failed", maybe this function shouldn't return an error at all and we should just log everything? I might just be overthinking this, I'm indifferent to any of the three options.